W-18077401 Anthropic MCP Connector 1.0 (#2835)
mulesoft/docs-connectors: mcp/1.0/modules/ROOT/pages/mcp-connector-reference.adoc

Configuration element that acts as a client and connects to an MCP server.

==== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
|Name | String | Name for this configuration. Connectors reference the configuration with this name. | | x
| Connection a| <<Client_SseClient, Sse Client>>
 | The connection types to provide to this configuration. | | x
| Client Name a| String |  Name the client uses to identify itself during the protocol initialization phase. This isn't the same as the name given to this config element (which is later referenced through config-ref parameters). This name doesn't need to be unique. |  | x
| Client Version a| String | Version the client uses to identify itself during the protocol initialization phase. This doesn't need to match the connector or Mule application version. |  | x
| Name a| String |  ID used to reference this configuration. |  | x
| Expiration Policy a| <<ExpirationPolicy>> |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before Mule considers it eligible for expiration.   |  | 
|===

==== Connection Types

[[Client_SseClient]]
===== SSE Client

> The SSE (Server-Sent Events) Client connection type enables MCP Connector to establish a persistent, oneway (server-sent) connection to an MCP server using the SSE standard. This type of connection allows the MCP server to push real-time updates and messages to the client without the need for the client to constantly poll for new information.

Reviewer: ```suggestion
The SSE (Server-Sent Events) Client connection type enables MCP Connector to establish a persistent, one-way (server-sent) connection to an MCP server using the SSE standard. This type of connection allows the MCP server to push real-time updates and messages to the client without the need for the client to constantly poll for new information.
```

|===

== Supported Operations

* <<CallTool>> 
* <<ListResources>> 
* <<ListTools>> 
* <<Ping>> 
* <<ReadResource>> 

[[Server]]
=== Server

Configuration element to act as an MCP server.

==== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
|Name | String | Name for this configuration. Connectors reference the configuration with this name. | | x
| Connection a| <<Server_SseServer, Sse Server>>
 | Connection types to provide to this configuration. | | x
| Server Name a| String |  Name this server uses to identify itself during the protocol initialization phase. This isn't the same as the name given to this config element (which is later referenced through config-ref parameters). This name doesn't need to be unique. |  | x
| Server Version a| String | Version this client uses to identify itself during the protocol initialization phase. This doesn't need to match the connector or Mule application version. |  | x
| Name a| String |  ID used to reference this configuration. |  | x

> | List Changed a| Boolean | Whether to enable notifications for changes to the tool list. |  true | 

Reviewer: List Changed field appears twice?

|===

[[CrlFile]]
=== CRL File

Specifies the location of the certification revocation list (CRL) file.

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Path a| String | Path to the CRL file. |  | 
|===

=== Reconnection

Configures a reconnection strategy for an operation.

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Fails Deployment a| Boolean | When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  | 
| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> | Reconnection strategy to use. |  | 
|===

[reconnect]]
=== Reconnect

[%header%autowidth.spread]
|===
| Field | Type | Description | Default Value | Required
| Frequency a| Number | How often in milliseconds to reconnect. | |
| Count a| Number | How many reconnection attempts to make. | |
| blocking |Boolean |If false, the reconnection strategy runs in a separate, non-blocking thread. |true |
|===

> [[reconnect-forever]]

Reviewer: ```suggestion
[[ReconnectForever]]
```

|===
| Field | Type | Description | Default Value | Required
| Initial Buffer Size a| Number | The amount of memory to allocate to consume the stream and provide random access to it. If the stream contains more data than can fit into this buffer, then the buffer expands according to the Buffer Size Increment value, with an upper limit of the Max In Memory Size value. instances. | 100 | 
| Buffer Size Increment a| Number | Amount by which the buffer size expands if it exceeds its initial size. Setting a value of `0` or lower specifies that the buffer can't expand. | 100 | 
| Max Buffer Size a| Number | Maximum size of the buffer. If the buffer size exceeds this value, Mule raises a `STREAM_MAXIMUM_SIZE_EXCEEDED` error. A value of less than or equal to `0` means no limit.  |  | 
|===

[[RepeatableFileStoreIterable]]
=== Repeatable File Store Iterable

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| In Memory Objects a| Number | Maximum number of instances to keep in memory. If more than the maximum is required, content on the disk is buffered. |  | 
| Buffer Unit a| Enumeration, one of:

** BYTE
** KB
** MB

> ** GB | Unit for the *In Memory Size* field. |  | 

Reviewer: ```suggestion
** GB | Unit for the *In Memory Objects* field. |  | 
```

|===

[[ToolMetadata]]
=== Tool Metadata

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Name a| String | Name of the tool. |  | x
| Description a| String |A detailed enough description for an LLM to determine when to call this tool. |  | x
| Input Schema a| String | JSON Schema describing the tool's input parameters.  |  | x
|===

[[NewSessionRequest]]
=== New Session Request

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Session Id a| String |  |  | x
| Additional Properties a| Object |  |  | x
|===

> 

Reviewer: ```suggestion
[[RedeliveryPolicy]]
```

=== Redelivery Policy

Configures the redelivery policy for executing requests that generate errors. You can add a redelivery policy to any source in a flow.

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Max Redelivery Count a| Number | Maximum number of times that a redelivered request can be processed unsuccessfully before returning a REDELIVERY_EXHAUSTED error. |  | 
| Message Digest Algorithm a| String | Secure hashing algorithm to use if the *Use Secure Hash* field is `true`. If the payload of the message is a Java object, Mule ignores this value and returns the value that the payload's `hashCode()` returned. |  | 
| Message Identifier a| <<RedeliveryPolicyMessageIdentifier>> | One or more expressions that determine if a message was redelivered. This property can be set only if the *Use Secure Hash* field is `false`. |  | 
| Object Store a| ObjectStore | Configures the object store that stores the redelivery counter for each message. |  | 
|===

> 

Reviewer: ```suggestion
[[RedeliveryPolicyMessageIdentifier]]
```
